zerver/lib/streams.py

```diff
@@ -461,9 +461,6 @@ def check_stream_access_for_delete_or_update(
     if user_profile.is_realm_admin:
         return
 
-    if sub is None and stream.invite_only:
-        raise JsonableError(error)
-
     if can_administer_accessible_channel(stream, user_profile):
```

Comment:
Hmm, this is now missing a check that the user has access to the channel before the `can_administer_accessible_channel` call. Probably further renaming will be appropriate, but it could be good to leave a comment to the effect of "Deleting a channel and most settings updates only require metadata access, not content access" if you're intentionally not doing such a check?